framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -203,6 +210,31 @@ MeshGenerator::declareMeshProperty(const std::string & data_name, const T & init
   return data;
 }
 
+template <typename T>
+T &
+MeshGenerator::setMeshProperty(const std::string & data_name, const T & data_value)
+{
+  if (!_app.executingMeshGenerators())
+    mooseError(
+        "Updating mesh meta data cannot occur in the constructor of mesh generators");
+
+  if (!hasMeshProperty(data_name, name()))
+    mooseError(
+        "Mesh meta data must have been initialized before calling setMeshProperty");
+
+  std::string full_name =
+      std::string(MeshMetaDataInterface::SYSTEM) + "/" + name() + "/" + data_name;
+
+  // Get metadata reference from RestartableDataMap and update its value
+  auto & restartable_data_map = _app.getRestartableDataMap(MooseApp::MESH_META_DATA);
+  auto iter = restartable_data_map.find(full_name);
+  RestartableData<T> & restartable_data_ref = static_cast<RestartableData<T> &>(*iter->second.value);
+  T & data = restartable_data_ref.set();
```

Comment:
this works but it would be cleaner to move this code to a getRestartableData routine?

Reply:
I created a function `getRestartableMetaData ` in MooseApp.C, since this class exposes RestartableData-related variables. Let me know if this function should be moved somewhere else instead.